x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/RemoteIndicesPermission.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.authz.permission;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xpack.core.security.authz.RestrictedIndices;
+import org.elasticsearch.xpack.core.security.authz.privilege.IndexPrivilege;
+import org.elasticsearch.xpack.core.security.support.Automatons;
+import org.elasticsearch.xpack.core.security.support.StringMatcher;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public record RemoteIndicesPermission(List<RemoteIndicesGroup> remoteIndicesGroups) {
+
+    public static final RemoteIndicesPermission NONE = new RemoteIndicesPermission(List.of());
+
+    public RemoteIndicesPermission forCluster(final String remoteClusterAlias) {
+        // TODO cache result
+        return new RemoteIndicesPermission(
+            remoteIndicesGroups.stream()
+                // TODO we can merge `indicesPermissionGroups` by `indices` here
```

Comment:
Functionally, merging here as the TODO suggests is not required. It would merely allows us to send a slightly more compact representation, when sending privs to the FC so leaving this as future work.